### PR TITLE
Port commonpp to Windows

### PR DESCRIPTION
**Pull request by 0x7f**
Works for me™ ;)

Versions I am using:
* Visual Studio 2017 Community Edition (built 64bit only)
* Boost 1.62.0 (from https://boost.teeks99.com/bin/1.62.0/boost_1_62_0-msvc-14.0-64.exe)
* Intel TBB 2017 Update 7 (https://github.com/01org/tbb/releases/download/2017_U7/tbb2017_20170604oss_win.zip)
* hwloc 1.11.7 (from https://www.open-mpi.org/software/hwloc/v1.11/downloads/hwloc-win64-build-1.11.7.zip)
* OpenSSL 1.0.2k (from https://www.conan.io/source/OpenSSL/1.0.2k/lasote/stable)
* libcurl 7.50.3 (from https://conan.io/source/libcurl/7.50.3/lasote/stable)

The TBB and HWLOC downloads include shared libs only. Did not try to used the ones from conan.io as well (see https://conan.io/source/hwloc/1.11.1/lasote/stable and https://www.conan.io/source/TBB/4.4.4/memsharded/testing). Maybe they allow static linking. Apart from that, all the other libraries were linked statically.

Here is the output of depends for the `shared_counter_bench.exe`:
image

Examples and tests are also running:
```

1>------ Build started: Project: RUN_TESTS, Configuration: Debug x64 ------
1>Test project C:/Users/max/Development/commonpp/build
1>    Start 1: core_options
1>1/7 Test #1: core_options .....................   Passed    0.02 sec
1>    Start 2: core_string_stringify
1>2/7 Test #2: core_string_stringify ............   Passed    0.02 sec
1>    Start 3: core_string_encode
1>3/7 Test #3: core_string_encode ...............   Passed    0.02 sec
1>    Start 4: metric_reservoir_edr
1>4/7 Test #4: metric_reservoir_edr .............   Passed    0.25 sec
1>    Start 5: net_http_encode-decode
1>5/7 Test #5: net_http_encode-decode ...........   Passed    0.04 sec
1>    Start 6: net_http_request
1>6/7 Test #6: net_http_request .................   Passed    0.03 sec
1>    Start 7: net_http_response
1>7/7 Test #7: net_http_response ................   Passed    0.03 sec
1>
1>100% tests passed, 0 tests failed out of 7
1>
1>Total Test time (real) =   0.43 sec
========== Build: 1 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========
```

### CODE REVIEW

File: src/commonpp/thread/detail/logger.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+*/
+
+#include "logger.hpp"
+
+namespace commonpp
+{
+namespace thread
+{
+
+DECLARE_LOGGER(thread_logger, "commonpp::thread");
```
Comment (0x7f) [inline src/commonpp/thread/detail/logger.cpp:19]: Not 100% sure whether it should be `DECLARE_LOGGER` or `CREATE_LOGGER` here. At least this patch fixed a linker error I received in httpp complaining about duplicate `thread_logger` symbol.

**Comment by daedric** [comment]
Hi @0x7f 
I'm surprised that you don't have to add the export/import stuff, in any case, I'll set up a windows CI so we can insure that further developments do not break the portability.

Thanks a lot for this work !

**Comment by daedric** [comment]
did some test, vcpkg works much better than conan :)
Will try to write a procedure as well as add commonpp to vcpkg. It'd allow httpp to depend on it thru vcpkg

**Comment by daedric** [comment]
I've added a commit on top of your branch in a local one: **https://github.com/daedric/commonpp/tree/port-to-windows**

I'm also currently trying to setup appveyor :)
I've had only one strange thing with VS2017 and I'll see if it reproduces on appveyor as well (running tests one by one works, but using the cmake builtin features of vs2017 will not link properly.)

**Pull request closed without merging**